oap-server/exporter/src/main/java/org/apache/skywalking/oap/server/exporter/provider/grpc/GRPCExporter.java

```diff
@@ -109,7 +109,7 @@ public void fetchSubscriptionList() {
                     lastFetchTimestamp = currentTimeMillis;
                     SubscriptionsResp subscription = blockingStub.withDeadlineAfter(10, TimeUnit.SECONDS)
                                                                  .subscription(SubscriptionReq.newBuilder().build());
-                    subscriptionList.clear();
+                    subscriptionList = new ArrayList<>();
                     subscriptionList.addAll(subscription.getMetricsList());
```

Comment:
This doesn't seem to fix either, `export` may be invoked right after `subscriptionList = new ArrayList<>()` but before `subscriptionList.addAll(subscription.getMetricsList());` (the lock doesn't guarantee these 2 operations to be all-or-none). Simply `subscriptionList = subscription.getMetricsList()` should do the trick without locks?

Reply:
Changed to `subscriptionList = subscription.getMetricsList()`. I still keep the lock to avoid repeated(and highly concurrently) changing the reference. What do you think?

Reply:
> Changed to `subscriptionList = subscription.getMetricsList()`. I still keep the lock to avoid repeated(and highly concurrently) changing the reference. What do you think?

Make sens to me 👍